core/iwasm/interpreter/wasm_loader.c

```diff
@@ -4190,6 +4177,38 @@ load_table_index(const uint8 **p_buf, const uint8 *buf_end, WASMModule *module,
     return false;
 }
 
+/* Element segments must match element type of table */
+static bool
+check_table_elem_type(WASMModule *module, uint32 table_index,
+                      uint32 type_from_elem_seg, char *error_buf,
+                      uint32 error_buf_size)
+{
+    uint32 table_declared_elem_type;
+
+    if (table_index < module->import_table_count)
+        table_declared_elem_type =
+            module->import_tables[table_index].u.table.elem_type;
+    else
+        table_declared_elem_type = (module->tables + table_index)->elem_type;
+
+    if (table_declared_elem_type == type_from_elem_seg)
+        return true;
+
+#if WASM_ENABLE_GC != 0
+    /*
+     * because load_table_segment_section() will
+     * process vec(funcidx) into (ref func) when mode is 0
+     * funcref == (ref func)
+     */
+    if (table_declared_elem_type == VALUE_TYPE_FUNCREF
+        && type_from_elem_seg == REF_TYPE_HT_NON_NULLABLE)
+        return true;
+#endif
```

Comment:
need a help here. 
- is the GC processing stable? 
- is it necessary to consider `NULL`?